tests/Unit/Shipping/LocationRatesProcessorTest.php

```diff
@@ -22,69 +21,88 @@ class LocationRatesProcessorTest extends UnitTest {
 	public function test_process_returns_only_most_expensive_flat_rate() {
 		$location       = new Location( 'US', 'CA' );
 		$location_rates = [
-			new LocationRate( $location, new ShippingRateFlat( 50 ) ),
-			new LocationRate( $location, new ShippingRateFlat( 100 ) ),
-			new LocationRate( $location, new ShippingRateFlat( 200 ) ),
+			new LocationRate( $location, new ShippingRate( 50 ) ),
+			new LocationRate( $location, new ShippingRate( 100 ) ),
+			new LocationRate( $location, new ShippingRate( 200 ) ),
 		];
-		$processed = $this->rates_processor->process( $location_rates );
+		$processed      = $this->rates_processor->process( $location_rates );
 		$this->assertEquals( 1, count( $processed ) );
-		/** @var ShippingRateFlat $flat_rate */
-		$flat_rate = $processed[0]->get_shipping_rate();
-		$this->assertInstanceOf( ShippingRateFlat::class, $flat_rate );
-		$this->assertEquals( 200, $flat_rate->get_rate() );
+		$this->assertEquals( 200, $processed[0]->get_shipping_rate()->get_rate() );
 	}
 
-	public function test_process_returns_only_highest_minimum_order_threshold_free_rate() {
-		$location       = new Location( 'US', 'CA' );
+	public function test_process_returns_only_highest_min_order_amount_free_rate() {
+		$location = new Location( 'US', 'CA' );
+
+		$free_rate_1 = new ShippingRate( 0 );
+		$free_rate_2 = new ShippingRate( 0 );
+		$free_rate_2->set_min_order_amount( 50 );
+		$free_rate_3 = new ShippingRate( 0 );
+		$free_rate_3->set_min_order_amount( 100 );
+		$free_rate_4 = new ShippingRate( 0 );
+		$free_rate_4->set_min_order_amount( 200 );
+
 		$location_rates = [
-			new LocationRate( $location, new ShippingRateFree() ),
-			new LocationRate( $location, new ShippingRateFree( 50 ) ),
-			new LocationRate( $location, new ShippingRateFree( 100 ) ),
-			new LocationRate( $location, new ShippingRateFree( 200 ) ),
+			new LocationRate( $location, new ShippingRate( 100 ) ), // We need a non-free rate so that the free rates are not ignored.
+			new LocationRate( $location, $free_rate_1 ),
+			new LocationRate( $location, $free_rate_2 ),
+			new LocationRate( $location, $free_rate_3 ),
+			new LocationRate( $location, $free_rate_4 ),
 		];
-		$processed = $this->rates_processor->process( $location_rates );
-		$this->assertEquals( 1, count( $processed ) );
-		/** @var ShippingRateFree $free_rate */
-		$free_rate = $processed[0]->get_shipping_rate();
-		$this->assertInstanceOf( ShippingRateFree::class, $free_rate );
-		$this->assertEquals( 200, $free_rate->get_threshold() );
+		$processed      = $this->rates_processor->process( $location_rates );
+		$this->assertEquals( 2, count( $processed ) );
+		foreach ( $processed as $location_rate ) {
+			if ( $location_rate->get_shipping_rate()->is_free() ) {
+				$this->assertEquals( 200, $location_rate->get_shipping_rate()->get_min_order_amount() );
```

Comment:
❓

Should not return 50 as the minum order amount?

Reply:
see https://github.com/woocommerce/google-listings-and-ads/pull/1422#discussion_r851109599

The same rule applies to the minimum order amount because we wouldn't want to set a low minimum order amount for a group if it only applies to a state or postcode within a country, etc.